File: QMP/QMPEQZ.pas
unit
  QMPEQZ;

interface

uses
  QMPEQU,
  QMPDCL;

type
  PQMPEQZ = ^TQMPEQZ;
  TQMPEQZ = record
  private
    class var fequ: TQMPEQU;
    class function Init(const Flags: Integer): Integer; cdecl; static;
    class function Modify(const Data: PData; const Latency: PInteger; const Flags: Integer): Integer; cdecl; static;
    class function Update(const Info: PInfo; const Flags: Integer): Integer; cdecl; static;
    class procedure Quit(const Flags: Integer); cdecl; static;
  public
    class function Plugin(): PPlugin; cdecl; static;
  end;

implementation

class function TQMPEQZ.Plugin(): PPlugin; cdecl;
begin
  Result := New(PPlugin);
  Result.Description := 'Quinnware Equalizer v3.51';
  Result.Version := $0000;
  Result.Init := TQMPEQZ.Init;
  Result.Quit := TQMPEQZ.Quit;
  Result.Modify := TQMPEQZ.Modify;
  Result.Update := TQMPEQZ.Update;
end;

class function TQMPEQZ.Init(const Flags: Integer): Integer; cdecl;
begin
  TQMPEQZ.fequ.Init();
  Result := 1;
end;

class procedure TQMPEQZ.Quit(const Flags: Integer); cdecl;
begin
  TQMPEQZ.fequ.Done();
end;

class function TQMPEQZ.Modify(const Data: PData; const Latency: PInteger; const Flags: Integer): Integer; cdecl;
begin
  TQMPEQZ.fequ.Process(Data);
  Result := 1;
end;

class function TQMPEQZ.Update(const Info: PInfo; const Flags: Integer): Integer; cdecl;
begin
  TQMPEQZ.fequ.Update(Info);
  Result := 1;
end;

begin
end.

File: CPP/qmpeqz.cpp
#pragma once
#include "qmpdcl.h"
#include "qmpdsp.h"
#include "qmpeqz.h"
#include "windows.h"

PPLUGIN CDECL QMPEQZ::plugin() {
    PPLUGIN result = new PLUGIN();

    result->description = L"Quinnware Equalizer v3.51";
    result->version = 0x0000;
    result->init = QMPEQZ::init;
    result->modify = QMPEQZ::modify;
    result->update = QMPEQZ::update;

    return result;
};

INT CDECL QMPEQZ::init(INT flags) {
    QMPEQZ::equ.init();

    return 1;
};

INT CDECL QMPEQZ::modify(PDATA data, PINT latency, INT flags) {
    QMPEQZ::equ.process(*data);

    return 1;
};

INT CDECL QMPEQZ::update(PINFO info, INT flags) {
    QMPEQZ::equ.update(*info);

    return 1;
};
